### PR TITLE
Backport monitor

### PR DESCRIPTION
**Pull request by jenshnielsen**
Backport Elm based monitor from QCoDeS DK

@jana-d

### CODE REVIEW

File: qcodes/__init__.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
 from qcodes.loops import Loop, active_loop, active_data_set
 from qcodes.measure import Measure
 from qcodes.actions import Task, Wait, BreakIf
+from qcodes.monitor.monitor import Monitor
```
Comment (jenshnielsen) [inline qcodes/__init__.py]: We might want to drop this import to not make websockets a hard dependency of qcodes?

**Comment by peendebak** [comment]
@jenshnielsen I just tested the PR from spyder. My system freezes after trying to start a monitor. I tried:
```
In [4]: m=qcodes.Monitor( gates.B0, gates.B1)
```
This results in the following error and my system being locked.
```
ERROR:asyncio:Exception in callback Monitor._create_task(<Future at 0x...state=pending>, <websockets.p...00211FE148AC8>)()
handle: <Handle Monitor._create_task(<Future at 0x...state=pending>, <websockets.p...00211FE148AC8>)()>
Traceback (most recent call last):
  File "C:\Users\eendebakpt\AppData\Local\Continuum\Anaconda3\envs\py35\lib\asyncio\events.py", line 125, in _run
    self._callback(*self._args)
  File "c:\svn\qcodes\qcodes\monitor\monitor.py", line 172, in _create_task
    task = self.loop.create_task(coro)
  File "C:\Users\eendebakpt\AppData\Local\Continuum\Anaconda3\envs\py35\lib\asyncio\base_events.py", line 261, in create_task
    task = tasks.Task(coro, loop=self)
  File "C:\Users\eendebakpt\AppData\Local\Continuum\Anaconda3\envs\py35\lib\asyncio\tasks.py", line 70, in __init__
    assert coroutines.iscoroutine(coro), repr(coro)
AssertionError: <websockets.py35.server.Serve object at 0x00000211FE148AC8>
```

**Comment by jenshnielsen** [comment]
I think this breaks with websockets 3.4 which is why setup.py limits it to 3.3. I initially missed that commit when backporting. Can you try with 3.3?

**Comment by peendebak** [comment]
@jenshnielsen Thanks, using websockets 3.3 solved the issue. Will setting of parameters also be made an option? That would be important for us to make it more usefull.

**Comment by jenshnielsen** [comment]
It's an old monitor that we have used and found useful in Copenhagen for some time which others have been interested in using hence this pull request. It's not meant for setting. We are aware of that request and it will happen eventually

**Comment by jenshnielsen** [comment]
@WilliamHPNielsen  added a readme with a link to the ELM repo

**Pull request merged**